mypy job in CI failing with mypy 1.1.1

The mypy CI job is now installing mypy 1.1.1 (previously was mypy 1.0.1), which appears to be flagging one error and one warning. Outputs from a mypy job are:

```
sympy/parsing/sympy_parser.py:1075: error: Incompatible types in assignment (expression has type "CodeType", variable has type "str")  [assignment]
sympy/ntheory/tests/test_qs.py:65: note: By default the bodies of untyped functions are not checked, consider using --check-untyped-defs  [annotation-unchecked]
```

--- a/sympy/ntheory/tests/test_qs.py
+++ b/sympy/ntheory/tests/test_qs.py
@@ -19,7 +19,7 @@ def test_qs_1():
         {18640889198609, 20991129234731}
 
 
-def test_qs_2():
+def test_qs_2() -> None:
     n = 10009202107
     M = 50
     # a = 10, b = 15, modified_coeff = [a**2, 2*a*b, b**2 - N]

--- a/sympy/parsing/sympy_parser.py
+++ b/sympy/parsing/sympy_parser.py
@@ -1072,7 +1072,7 @@ def parse_expr(s: str, local_dict: Optional[DICT] = None,
     code = stringify_expr(s, local_dict, global_dict, _transformations)
 
     if not evaluate:
-        code = compile(evaluateFalse(code), '<string>', 'eval')
+        code = compile(evaluateFalse(code), '<string>', 'eval') # type: ignore
 
     try:
         rv = eval_expr(code, local_dict, global_dict)

